src/effects/VST/VSTEffect.cpp

```diff
@@ -1207,8 +1207,9 @@ int VSTEffect::ShowClientInterface(
    return mDialog->ShowModal();
 }
 
-bool VSTEffect::SaveSettings(
-   const EffectSettings &, CommandParameters & parms) const
+
+// This used to be ::SaveSettings
```

Comment:
Okay, commit 6.2, the most difficult one:

I'm not convinced by what you write in the commit comment, about the need to keep the old functions.  And, the new LoadSettings/SaveSettings are not using the same "parm_%d" convention to name the keys (sometimes), which means it might break compatibility with old config files.

`StoreCommandParameters` is used only by `LoadParameters`.  If it takes the "chunk" alternative, then it is using the mAEffect handle, as a "scratch" from which it then fetches settings.  That's ok.

If that function does not take the early exit path loaing a chunk -- then instead a string value for key "Parameters" is loaded, and reinterpreted as a config file of key-value pairs.  I don't see why this could not be given to the new `LoadSettings` directly.  In which case, you do not rely on a "scratch" handle and so should NOT use `FetchSettings`.

`FetchCommandParameters` is used only by `SaveParameters`.  That function first calls `StoreSettings` to change the scratch handle state, and then maybe exits early if chunks are supported.  But if they are not, then again, I don't understand why the direct translation of settings to key-value pairs in `eap` can't be done.

Reply:
I interpret your comment like so: in `LoadParameters`, instead of calling `StoreCommandParameters` and then `FetchSettings`, call `LoadSettings` instead. Then there would be no need for StoreCommandParameters to exist.

But the thing is: the new LoadSettings only translates to settings the CommandParameters that it is given, without writing them to the handle. I thought this was the semantics of LoadSettings we agreed on - is it not? it should be only a translator, not a translator + a setter. 

Because I found the existing LoadSettings not to conform to these agreed semantics (it was also writing to the handle), I renamed it to StoreCommandParameters and wrote a new LoadSettings which does the translation only.

But I could also revert it all and just have the original LoadSettings which translates and sets to the handle.  No problem for me, the code would also be simpler. I would only have the concern that the semantics of LoadSettings vary across effect families.